e2e.go

```diff
@@ -208,94 +196,53 @@ func (a *Asserter) Errorf(format string, args ...interface{}) {
 	a.Err = fmt.Errorf(format, args...)
 }
 
-func QueryNetworkState(e2eConfig *E2EConfig) (LatestLedgerResult, error) {
-	getLatestLedger := []byte(`{
-           "jsonrpc": "2.0",
-           "id": 10235,
-           "method": "getLatestLedger"
-        }`)
-
-	resp, err := http.Post(e2eConfig.TargetNetworkRPCURL, "application/json", bytes.NewBuffer(getLatestLedger))
-	if err != nil {
-		return LatestLedgerResult{}, fmt.Errorf("soroban rpc get latest ledger had error %e", err)
-	}
-
-	var rpcResponse RPCLatestLedgerResponse
-	decoder := json.NewDecoder(resp.Body)
-	err = decoder.Decode(&rpcResponse)
-	if err != nil {
-		return LatestLedgerResult{}, fmt.Errorf("soroban rpc get latest ledger, not able to parse response, %v, %e", resp, err)
-	}
-	if rpcResponse.Error != nil {
-		return LatestLedgerResult{}, fmt.Errorf("soroban rpc get latest ledger, error on response, %v, %e", resp, err)
-	}
-
-	return rpcResponse.Result, nil
-
+func QueryNetworkState(e2eConfig *E2EConfig) (protocol.GetLatestLedgerResponse, error) {
+	cli := client.NewClient(e2eConfig.TargetNetworkRPCURL, nil)
```

Comment:
nice refactor to the sdk!

Reply:
ugh thank you but I committed that by accident - I wanted it to be separate so I could isolate test failures vs. code changes :cry: thanks for the callout because otherwise I wouldn't have noticed lol - I'll separate this change!